lib/api/stack.ts

```diff
@@ -93,8 +98,49 @@ export class ApiStack extends cdk.Stack {
       responseMappingTemplate: appsync.MappingTemplate.dynamoDbResultList(),
     });
 
+    recipientTableDataSource.createResolver({
+      typeName: "Query",
+      fieldName: "listRecipients",
+      requestMappingTemplate: appsync.MappingTemplate.fromString(
+        `
+        {
+          "version" : "2017-02-28",
+          "operation" : "Query",
+          "query" : {
+            "expression": "tenant = :tenant",
+              "expressionValues" : {
+                ":tenant" : $util.dynamodb.toDynamoDBJson($context.identity.claims.get("cognito:groups").get(0))
+              }
+          }
+        }
+      `
+      ),
+      responseMappingTemplate: appsync.MappingTemplate.dynamoDbResultList(),
+    });
+
+    recipientTableDataSource.createResolver({
+      typeName: "Mutation",
+      fieldName: "addRecipient",
+      requestMappingTemplate: appsync.MappingTemplate.fromString(
+        `
+        {
+          "version" : "2017-02-28",
+          "operation" : "PutItem",
+          "key" : {
+            "tenant": $util.dynamodb.toDynamoDBJson($context.identity.claims.get("cognito:groups").get(0)),
+            "address": $util.dynamodb.toDynamoDBJson($context.arguments.input.address)
+          },
+          "attributeValues": {
+            "pools" : $util.dynamodb.toDynamoDBJson($context.arguments.input.pools),
```

Comment:
With the way this is updated the frontend needs to take care of providing the full list of associated pools on an update request, right? When trying to imagine a dialog that would support this I'm wondering if it really fits the pool list management workflow I would prefer as a user.